### PR TITLE
Added sound for player movement, Button sounds, level transition sound.

### PR DESCRIPTION
**Pull request by eddR6**

### CODE REVIEW

**Comment by eddR6** [comment]
How to reset vol,pitch etc after sound play?

File: Assets/Scripts/SoundManager.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+public class SoundManager : MonoBehaviour
+{
+    private static SoundManager instance;
+    public static SoundManager Instance { get { return instance; } }
+    public SoundType[] SoundMap;
+    public AudioSource soundEffect;
+    public AudioSource soundMusic;
+    private void Awake()
+    {
+        if (instance == null)
```
Comment (mayankgrover) [inline Assets/Scripts/SoundManager.cs:15]: Debug.Log()
Debug.LogError()
Debug.LogWarning()

File: Assets/Scripts/SoundManager.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    public void Play(Sounds sound)
+    {
+        AudioClip clip = GetSoundClip(sound);
+        if (clip != null)
+        {
+            soundEffect.PlayOneShot(clip);
+        }
+        else
+        {
+            Debug.Log("Clip no found.");
```
Comment (mayankgrover) [inline Assets/Scripts/SoundManager.cs]: warning

File: Assets/Scripts/SoundManager.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        }
+    }
+    public void PlayMusic(Sounds sound)
+    {
+        AudioClip soundClip = GetSoundClip(sound);
+        soundMusic.clip = soundClip;
+        soundMusic.volume = 0.4f;
```
Comment (mayankgrover) [inline Assets/Scripts/SoundManager.cs]: magic numbers

File: Assets/Scripts/SoundManager.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+[Serializable]
+public class SoundType
+{
+    public Sounds soundType;
+    public AudioClip soundClip;
+}
+
+public enum Sounds
+{
+    ButtonClick,
```
Comment (mayankgrover) [inline Assets/Scripts/SoundManager.cs:77]: None

Comment (mayankgrover) [reply Assets/Scripts/SoundManager.cs:77]: Default

File: Assets/Scripts/SoundManager.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        AudioClip soundClip = GetSoundClip(sound);
+        soundMusic.clip = soundClip;
+        soundMusic.volume = 0.4f;
+        soundMusic.Play();
+    }
+
+    private AudioClip GetSoundClip(Sounds sound)
+    {
+        return Array.Find(SoundMap, item => item.soundType == sound).soundClip;
```
Comment (mayankgrover) [inline Assets/Scripts/SoundManager.cs]: if(sounds == Sounds.None) return null; 

return Array.Find(SoundMap, item => item.soundType == sound).soundClip;

File: Assets/Scripts/SoundManager.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    }
+}
+[Serializable]
+public class SoundType
+{
+    public Sounds soundType;
+    public AudioClip soundClip;
```
Comment (mayankgrover) [inline Assets/Scripts/SoundManager.cs:69]: public float volume;

Comment (mayankgrover) [reply Assets/Scripts/SoundManager.cs:69]: public bool isLooping;

**Review by mayankgrover — CHANGES_REQUESTED** [review]

**Pull request merged**